beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2875,6 +2886,58 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             );
         }
 
+        // Do not write to eth1 data cache for blocks older than 5 epochs, this helps reduce
+        // noise during sync
+        if block_delay_total < self.slot_clock.slot_duration() * 160 {
+            let parent_block_epoch = fully_verified_block
+                .parent_block
+                .slot()
+                .epoch(T::EthSpec::slots_per_epoch());
+            if parent_block_epoch < current_epoch {
+                // we've crossed epoch boundary, store Eth1CacheData
+                let (checkpoint, eth1_cache_data) =
+                    if current_slot % T::EthSpec::slots_per_epoch() == 0 {
+                        // current block is the checkpoint
+                        (
+                            Checkpoint {
+                                epoch: current_epoch,
+                                root: block_root,
+                            },
+                            current_eth1_cache_data,
+                        )
+                    } else {
+                        // parent block is the checkpoint
+                        (
+                            Checkpoint {
+                                epoch: current_epoch,
+                                root: fully_verified_block.parent_block.canonical_root(),
+                            },
+                            fully_verified_block.parent_eth1_cache_data,
+                        )
+                    };
+
+                if let Some(Some(finalized_eth1_data)) = self
```

Comment:
We can avoid the double `Some` by using `and_then` below

```suggestion
                if let Some(finalized_eth1_data) = self
```